server/api_test.go

```diff
@@ -968,3 +968,88 @@ func TestGetConnectedChannels(t *testing.T) {
 		})
 	}
 }
+
+func TestMSTeamsTeamList(t *testing.T) {
+	for _, test := range []struct {
+		Name               string
+		SetupPlugin        func(*plugintest.API)
+		SetupStore         func(*storemocks.Store)
+		SetupClient        func(*clientmocks.Client)
+		ExpectedResult     string
+		ExpectedStatusCode int
+	}{
+		{
+			Name:        "MSTeamsTeamList: MS Teams team listed successfully",
+			SetupPlugin: func(api *plugintest.API) {},
+			SetupStore: func(store *storemocks.Store) {
+				store.On("GetTokenForMattermostUser", testutils.GetUserID()).Return(&oauth2.Token{}, nil).Times(2)
```

Comment:
I think we should change our code in such a way that this is called only once. Because once we have called this, we can make the client, then what's the need for calling again?